backend/src/usersAvatar/services/usersAvatar.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, InternalServerErrorException } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { ConfigService } from "@nestjs/config";
+import { UserAvatar } from "../model/usersAvatar.entity";
+import { UsersService } from "users/services/users.service";
+import { unlinkSync } from "fs";
+import { dirname, join } from "path";
+
+@Injectable()
+export class UsersAvatarService {
+    constructor(
+        @InjectRepository(UserAvatar)
+        private readonly usersAvatarRepository: Repository<UserAvatar>,
+        private readonly configService: ConfigService,
+        private readonly usersService: UsersService
+    ) {}
+
+    async addUserAvatarFile(id: number, file: Express.Multer.File) {
+        const name = file.originalname;
+        const extension = file.originalname.split(".").pop();
+        const sourcePath = file.path;
+
+        const { userAvatar } = await this.usersService.findOneById(id);
+
+        userAvatar && (await this.remove(userAvatar.id, userAvatar.sourcePath));
```

Comment:
No side effects. It should be divided in separated functions or change function name.

Reply:
what, will be divided?

Reply:
`await this.remove(userAvatar.id, userAvatar.sourcePath)` should be done in separated method not there. Because function name is "addUserAvatarFile" so it should be only addUserAvatar because there is no mention about removing something. Or rename name function to something other that will give information that there is also deleting stuff.

Reply:
I didn't check if image exist, because it's take more resources, over it i remove file and add 1 more time, this remove is in other function (remove) in the same service, your req changed is not neccesary